sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OpenTelemetrySdkAutoConfiguration.java

```diff
@@ -50,5 +61,26 @@ private static void configureMeterProvider(Resource resource, ConfigProperties c
     MetricExporterConfiguration.configureExporter(exporterName, config, meterProvider);
   }
 
+  private static Resource buildResource() {
+    ConfigProperties config = ConfigProperties.get();
+    Resource result = Resource.getDefault();
+
+    // TODO(anuraaga): We use a hyphen only once in this artifact, for
+    // otel.java.disabled.resource-providers. But fetching by the dot version is the simplest way
+    // to implement it for now.
+    Set<String> disabledProviders =
+        new HashSet<>(config.getCommaSeparatedValues("otel.java.disabled.resource.providers"));
+    for (ResourceProvider resourceProvider : ServiceLoader.load(ResourceProvider.class)) {
+      if (disabledProviders.contains(resourceProvider.getClass().getName())) {
+        continue;
+      }
+      result = result.merge(resourceProvider.createResource(config));
+    }
+
+    result = result.merge(EnvironmentResource.getInstance());
```

Comment:
`EnvironmentResource` does not really need a static final instance anymore, so if you just make it a simple factory class you'll be able to reuse `config` and avoid constructing another `ConfigProperties` inside the `EnvironmentResource` class